File: lib/FacultyAPR.Models/FacultyAPR.Models/Users/FacultyUser.cs
using System;

namespace FacultyAPR.Models
{
    public class FacultyUser: IUser
    {
        public Guid? Id { get; set; }
        
        public string FirstName { get; set; }
        
        public string LastName { get; set; }
        
        public string EmailAddress { get; set;}

        public FacultyRank Rank { get; set; }
        
        public CollegeDepartment Department { get; set; }

        public UserType UserType { get; set;}
    }
}

File: FacultyAPR.Client/src/models/Users/FacultyUser.ts
import IUser from "./IUser"
import Rank from "../BusinessObjects/Rank"
import Department from "../BusinessObjects/Department"
import UserType from "./UserType"

type FacultyUser = IUser & {
    rank: Rank;
    department: Department;
    userType: UserType.Faculty;
};
// Example usage
// let user = {} as FacultyUser

export default FacultyUser